erpnext/controllers/status_updater.py

```diff
@@ -167,7 +167,7 @@ def validate_completed_qty_for_row(self, row, completed_field, reference_field,
 			difference = completed_qty - reference_qty
 			excess_qty = difference
 		elif allowance_type == "max_qty_field":
-			max_qty = flt(row.get(max_qty_field))
+			max_qty = flt(row.get(max_qty_field), row.precision(max_qty_field))
```

Comment:
Not mandatory. Also preferable to use precision of reference_field for consistency in rounding